src/psse_export.jl

```diff
@@ -0,0 +1,1369 @@
+const PSSE_EXPORT_SUPPORTED_VERSIONS = [:v33]  # TODO add :v34
+const PSSE_DEFAULT = ""  # Used below in cases where we want to insert an empty field to signify the PSSE default
+const PSSE_BUS_TYPE_MAP = Dict(
+    PSY.ACBusTypes.PQ => 1,
+    PSY.ACBusTypes.PV => 2,
+    PSY.ACBusTypes.REF => 3,
+    PSY.ACBusTypes.SLACK => 3,
+    PSY.ACBusTypes.ISOLATED => 4,
+)
+
+# Each of the groups in the PSS/3 v33 standard
+const PSSE_GROUPS_33 = [
+    "Case Identification Data",
+    "Bus Data",
+    "Load Data",
+    "Fixed Shunt Data",
+    "Generator Data",
+    "Non-Transformer Branch Data",
+    "Transformer Data",
+    "Area Interchange Data",
+    "Two-Terminal DC Transmission Line Data",
+    "Voltage Source Converter (VSC) DC Transmission Line Data",
+    "Transformer Impedance Correction Tables",
+    "Multi-Terminal DC Transmission Line Data",
+    "Multi-Section Line Grouping Data",
+    "Zone Data",
+    "Interarea Transfer Data",
+    "Owner Data",
+    "FACTS Device Data",
+    "Switched Shunt Data",
+    "GNE Device Data",
+    "Induction Machine Data",
+    "Q Record",
+]
+
+const PSSE_DEFAULT_EXPORT_NAME = "export"
+
+const PSSE_RAW_BUFFER_SIZEHINT = 1024
+const PSSE_MD_BUFFER_SIZEHINT = 1024
+
+# TODO move this to IS
+"""
+A context manager similar to `Logging.with_logger` that sets the system's units to the given
+value, executes the function, then sets them back. Suppresses logging below `Warn` from
+internal calls to `set_units_base_system!`. Not thread safe.
+"""
+function with_units(f::Function, sys::System, units::Union{PSY.UnitSystem, String})
+    old_units = PSY.get_units_base(sys)
+    Logging.with_logger(Logging.SimpleLogger(Logging.Warn)) do
+        PSY.set_units_base_system!(sys, units)
+    end
+    try
+        f()
+    finally
+        Logging.with_logger(Logging.SimpleLogger(Logging.Warn)) do
+            PSY.set_units_base_system!(sys, old_units)
+        end
+    end
+end
+
+"""
+Make a `deepcopy` of the `System` except replace the time series manager and supplemental
+attribute manager with blank versions so these are not copied.
+"""
+function deepcopy_system_no_time_series_no_supplementals(sys::System)
```

Comment:
@GabrielKS  and @daniel-thom  should we move this PSY and make a "fast" copy function. Despite the fact that we might move to some other approach, this seems useful in other areas like the saving of the system to results in PowerSimulations.jl

Reply:
(see previous discussion at https://github.com/NREL-Sienna/PowerFlows.jl/pull/48#discussion_r1819901045)

Reply:
Sure, I agree. I do wonder if someone is going to want an option that keeps supplemental attributes.

Reply:
Maybe it's `fast_deepcopy_system` with kwargs `include_time_series` and `include_supplemental_attributes` or something like that

Reply:
-> https://github.com/NREL-Sienna/PowerSystems.jl/issues/1231